forc-plugins/forc-client/src/util/tx.rs

```diff
@@ -121,13 +129,65 @@ impl<Tx: Buildable + SerializableVec + field::Witnesses + Send> TransactionBuild
         client: FuelClient,
         unsigned: bool,
         signing_key: Option<SecretKey>,
+        wallet_mode: WalletSelectionMode,
     ) -> Result<Tx> {
         let params = client.chain_info().await?.consensus_parameters.into();
         let mut signature_witness_index = 0u8;
-        if !unsigned {
+        let signing_key = if !unsigned {
+            let key = match (wallet_mode, signing_key) {
+                (WalletSelectionMode::ForcWallet, None) => {
+                    // TODO: This is a very simple TUI, we should consider adding a nice TUI
+                    // capabilities for selections and answer collection.
+                    let wallet_path = default_wallet_path();
+                    if !wallet_path.exists() {
+                        anyhow::bail!("Cannot find a wallet at {wallet_path:?}\nPlease a generate new wallet with `forc wallet new`");
+                    }
+                    let prompt = format!(
+                        "\nPlease provide the password of your encrypted wallet vault at {wallet_path:?}:"
+                    );
+                    let password = rpassword::prompt_password(prompt)?;
+                    // TODO: List all derived wallets via forc-wallet and let the users choose
```

Comment:
nit: if we have issues for any of these TODOs, we should reference the issue number in the comment.

Reply:
We do not have them currently. I will be opening them tomorrow morning and will be tackling these todos as the first thing in the next PR